numfun/funutils.py

```diff
@@ -90,20 +90,21 @@ def detect_edge(f, domain=np.array([-1.0, 1.0]), v_scale=1.0, h_scale=2.0) -> fl
 
         if len(nz) == 0:
             # Derivatives are not growing;
-            return np.zeros((0,))
+            return cast(float, np.zeros((0,)))
```

Comment:
we want to make sure that the output is an np array. Can you please explain what does `cast` do? how is it useful?

Reply:
`cast` does not do anything, this is its implementation:
```python
def cast(typ, val):
    """Cast a value to a type.

    This returns the value unchanged.  To the type checker this
    signals that the return value has the designated type, but at
    runtime we intentionally don't check anything (we want this
    to be as fast as possible).
    """
    return val
```
Why use it then? It basically tells the type checker "hey type checker i know better than you what the type of this variable is" or "hey type checker you cannot infer what type the variable has so i will tell you explicitly". This function returns a float number as a result of `np.mean` **OR** an empty `ndarray`.